app/models/user_auth_token.rb

```diff
@@ -26,6 +30,27 @@ class UserAuthToken < ActiveRecord::Base
     )
   end
 
+  def user
+    impersonated_user || super
```

Comment:
```suggestion
    impersonated_user || acting_user
```

Tiny suggestion to make things clearer.